### PR TITLE
Correctly build packages for linux-musl-arm64

### PR DESCRIPTION
**Pull request by jashook**
Also add build-package step to PR.

### CODE REVIEW

File: eng/build-job.yml [hunk 1]
```diff
@@ -140,12 +140,17 @@ jobs:
           KeyVaultName: EngKeyVault
           SecretsFilter: 'dotnetfeed-storage-access-key-1,microsoft-symbol-server-pat,symweb-symbol-server-pat'
 
+    - ${{ if ne(parameters.osGroup, 'Windows_NT') }}:
+      - script: ./build-packages.sh -BuildArch=$(archType) -BuildType=$(buildConfigUpper) $(crossPackagesArg) -OfficialBuildId=$(Build.BuildNumber) $(portableBuildArg)
+        displayName: Build packages
+
```
Comment (sbomer) [inline eng/build-job.yml]: Nit: could you remove the line between the build-packages.sh/cmd steps and add a comment above the two to make it visually clear that they're conceptually the same step?

Comment (jashook) [reply eng/build-job.yml]: Sure

**Comment by jashook** [comment]
Look at that its already working :\

**Comment by jashook** [comment]
Investigating....

**Comment by jashook** [comment]
Fixed with a hack.

**Comment by jashook** [comment]
Change passed all builds, this will not effect testing merging to free machines.

**Pull request merged**